src/main/java/com/gdschongik/gdsc/domain/email/domain/HongikUnivEmailValidator.java

```diff
@@ -23,4 +24,18 @@ public void validateSendUnivEmailVerificationLink(String email, boolean isUnivEm
             throw new CustomException(UNIV_EMAIL_ALREADY_SATISFIED);
         }
     }
+
+    /**
+     * redis 안의 존재하는 메일인증 정보로 검증
+     * 1. 토큰이 비었는데 인증하려할 시 에러 (인증메일을 보내지 않았거나, 만료된 경우)
+     * 2. 토큰이 redis에 저장된 토큰과 다르면 만료되었다는 에러 (메일 여러번 보낸 경우)
+     */
+    public void validateUnivEmailVerification(
+            Optional<UnivEmailVerification> optionalUnivEmailVerification, String currentToken) {
+        if (optionalUnivEmailVerification.isEmpty()) {
+            throw new CustomException(EMAIL_NOT_SENT);
+        } else if (!optionalUnivEmailVerification.get().getVerificationToken().equals(currentToken)) {
```

Comment:
style이긴 한데 뒤따르는 if문의 조건이 앞의 조건의 영향을 받지 않는다면 대체로 else if 보단 별도의 if를 사용하고 있습니다!
```suggestion
        } 
        
        if (!optionalUnivEmailVerification.get().getVerificationToken().equals(currentToken)) {
```